torchvision/datasets/utils.py

```diff
@@ -104,7 +104,11 @@ def _get_google_drive_file_id(url: str) -> Optional[str]:
 
 
 def download_url(
-    url: str, root: str, filename: Optional[str] = None, md5: Optional[str] = None, max_redirect_hops: int = 3
+    url: str,
+    root: Union[str, bytes, os.PathLike],
```

Comment:
This can probably just be:

```suggestion
    root: Union[str, pathlib.Path]
```